go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -274,53 +280,28 @@ func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.Egress
 	}
 
 	ipv4HashedAS, ipv6HashedAS := nsInfo.addressSet.GetASHashNames()
-	err = oc.addEgressFirewallRules(ipv4HashedAS, ipv6HashedAS, egressFirewall.Namespace, egressFirewallStartPriorityInt)
+	err = oc.addEgressFirewallRules(ipv4HashedAS, ipv6HashedAS, egressFirewall.Namespace, egressFirewallStartPriorityInt, txn)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall) error {
-	// block all external traffic in this namespace
-	nsInfo, err := oc.waitForNamespaceLocked(newEgressFirewall.Namespace)
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
+func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall, txn *util.NBTxn) error {
+	updateErrors := oc.deleteEgressFirewall(oldEgressFirewall, txn)
+	if updateErrors != nil {
+		return updateErrors
 	}
-	addressSet := nsInfo.addressSet
-	nsInfo.Unlock()
-	priority, err := strconv.Atoi(types.EgressFirewallStartPriority)
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
-	}
-
-	ipv4ASHashName, ipv6ASHashName := addressSet.GetASHashNames()
-	match := generateMatch(ipv4ASHashName, ipv6ASHashName, []matchTarget{
-		{matchKindV4CIDR, "0.0.0.0/0"},
-		{matchKindV6CIDR, "::/0"},
-	},
-		[]egressfirewallapi.EgressFirewallPort{},
-	)
-
-	err = oc.createEgressFirewallRules(priority, match, "drop", newEgressFirewall.Namespace+"-blockAll")
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
+	updateErrors = oc.addEgressFirewall(newEgressFirewall, txn)
```

Comment:
@JacobTanenbaum can we just `return oc.addEgressFirewall(newEgressFirewall, txn)`? Otherwise the code is:

```
if updateErrors != nil {
    return updateErrors
}
return updateErrors
```

Reply:
done